scio-core/src/main/scala/com/spotify/scio/VersionUtil.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2016 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio
+
+import com.google.api.client.http.javanet.NetHttpTransport
+import com.google.api.client.http.{GenericUrl, HttpRequest, HttpRequestInitializer}
+import com.google.api.client.json.JsonObjectParser
+import com.google.api.client.json.jackson2.JacksonFactory
+import org.slf4j.LoggerFactory
+
+import scala.collection.mutable
+import scala.util.Try
+
+private object VersionUtil {
+
+  case class SemVer(major: Int, minor: Int, rev: Int, snapshot: Int) {
+    def toInt: Int = major * 1000000 + minor * 10000 + rev * 100 + snapshot
```

Comment:
I understand how practically this is gonna work - but in theory this breaks. Counterargument:

```
my-version: 0.1.101 (0 + 10000 + 10100 = 20100)
release: 0.2.0 (0 + 20000 + 0 = 20000)
```

I think you could use Algebird's `Min`:

```
import com.twitter.algebird._
import com.twitter.algebird.Operators._
Min((0,2,0,"")) + Min((0,1,101,"")) -> Min((0,1,101,))
Min((0,2,0,"")) + Min((0,2,0,"SNAPSHOT")) -> Min((0,2,0,))
```